language/tools/move-package/tests/test_runner.rs

```diff
@@ -2,19 +2,61 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use move_command_line_common::testing::{format_diff, read_env_update_baseline, EXP_EXT};
-use move_package::source_package::manifest_parser as MP;
-use std::{fs, path::Path};
+use move_package::{
+    compilation::build_plan::BuildPlan, resolution::resolution_graph as RG,
+    source_package::manifest_parser as MP, BuildConfig,
+};
+use std::{
+    ffi::OsStr,
+    fs,
+    path::{Component, Path, PathBuf},
+};
+
+const COMPILE_EXT: &str = "compile";
 
 pub fn run_test(path: &Path) -> datatest_stable::Result<()> {
     let update_baseline = read_env_update_baseline();
+    if path
+        .components()
+        .any(|component| component == Component::Normal(OsStr::new("deps_only")))
+    {
+        return Ok(());
+    }
     let exp_path = path.with_extension(EXP_EXT);
+    let should_compile = path.with_extension(COMPILE_EXT).is_file();
 
     let exp_exists = exp_path.is_file();
 
     let contents = fs::read_to_string(path)?;
-    let output = match MP::parse_move_manifest_string(contents).and_then(MP::parse_source_manifest)
+    let output = match MP::parse_move_manifest_string(contents)
+        .and_then(MP::parse_source_manifest)
+        .and_then(|parsed_manifest| {
+            RG::ResolutionGraph::new(
+                parsed_manifest,
+                path.parent().unwrap().to_path_buf(),
+                BuildConfig { dev_mode: true },
+            )
+        })
+        .and_then(|rg| rg.resolve())
     {
-        Ok(parsed_package) => format!("{:#?}\n", parsed_package),
+        Ok(mut resolved_package) => {
+            if should_compile {
+                match BuildPlan::create(resolved_package).and_then(|bp| bp.compile()) {
+                    Ok(mut pkg) => {
+                        pkg.compiled_package_info.source_digest =
+                            Some("ELIDED_FOR_TEST".to_string());
+                        format!("{:#?}", pkg.compiled_package_info)
```

Comment:
I believe adding a `\n` to the end of this format string, or to the call to `fs::write` below, ought to result in newlines at the end of the expectation files.

Reply:
Good spot -- I keep forgetting to add them here!